### PR TITLE
Avoid duplicate derives

### PR DESCRIPTION
**Pull request by jayvdb**
If code was using `with_derive("Foo")`, and `Foo` is added to the base derive list, the result is a duplicate derive.

This prevents that from being a failure, and also sorts the combined derives.

### CODE REVIEW

**Review by ahl — COMMENTED** [review]
seems good; can you please resolve conflicts?

**Comment by jayvdb** [comment]
@ahl , PR rebased.  Workflow needs approval again. Thanks.

**Comment by jayvdb** [comment]
@ahl this is green.  Any other issues, or can it be merged?

**Comment by jayvdb** [comment]
ping @ahl

File: typify-impl/src/type_entry.rs [hunk 1]
```diff
@@ -1232,15 +1232,14 @@ fn strings_to_derives<'a>(
...
-                .into_token_stream()
-        })
+    let mut combined_derives = derive_set.clone();
+    combined_derives.extend(extra_derives.iter().map(String::as_str));
+    combined_derives.extend(type_derives.iter().map(String::as_str));
+    combined_derives.into_iter().map(|derive| {
+        syn::parse_str::<syn::Path>(derive)
+            .unwrap()
+            .into_token_stream()
+    })
```
Comment (ahl) [inline typify-impl/src/type_entry.rs:1242]: what's the purpose of this change?

**Review by ahl — COMMENTED** [review]
looking good; one question

File: typify-impl/src/type_entry.rs (hunk 1 above)
Comment (jayvdb) [reply typify-impl/src/type_entry.rs:1242]: This change builds a (new) single set of all of the derives, thereby eliminating duplicates.
